### PR TITLE
Get VRC and verify driver

### PR DESCRIPTION
**Pull request by iamniting**
Signed-off-by: Nitin Goyal <nigoyal@redhat.com>

### CODE REVIEW

**Comment by iamniting** [comment]
duplicate of https://github.com/kube-storage/volume-replication-operator/pull/23

File: controllers/volumereplication_controller.go [hunk 1]
```diff
@@ -52,7 +56,72 @@ type VolumeReplicationReconciler struct {
...
+		if errors.IsNotFound(err) {
+			r.Log.Info("VolumeReplication not found", "object", req.NamespacedName)
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	// Get volume replication class
+	vrcObj := &replicationv1alpha1.VolumeReplicationClass{}
+	err = r.Client.Get(context.TODO(), types.NamespacedName{Name: vrObj.Spec.VolumeReplicationClass}, vrcObj)
```
Comment (Madhu-1) [inline controllers/volumereplication_controller.go:83]: question: will this work for clusterScoped resources?

File: controllers/volumereplication_controller.go [hunk 2]
```diff
@@ -52,7 +56,72 @@ type VolumeReplicationReconciler struct {
...
+			r.Log.Error(err, "VolumeReplicationClass not found", "object", vrObj.Spec.VolumeReplicationClass)
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, err
+	}
+
+	if r.DriverConfig.DriverName != vrcObj.Spec.Provisioner {
+		err = fmt.Errorf("VolumeReplicationClass Provisioner %s does not match with %s", vrcObj.Spec.Provisioner, r.DriverConfig.DriverName)
+		r.Log.Error(err, "Provisioner does not match")
+		return ctrl.Result{}, err
```
Comment (Madhu-1) [inline controllers/volumereplication_controller.go:95]: As the driver name is not matching we should not log or return an error we just need to return success here

**Comment by Madhu-1** [comment]
Merge this after merging #23

**Review by sp98 — APPROVED** [review]

**Review by Madhu-1 — APPROVED** [review]

**Pull request merged**